internal/processor/parser/util.go

```diff
@@ -21,4 +22,14 @@ func appendTagIfNotEmpty(tags []string, key, value string) []string {
 		tags = append(tags, fmt.Sprintf("%s:%s", key, value))
 	}
 	return tags
+}
+
+func getContainerInstanceID(gauge *loggregator_v2.Gauge, instanceID string) string {
+	if gauge == nil {
+		return ""
+	}
+	if id, ok := gauge.GetMetrics()["instance_index"]; ok && id != nil {
```

Comment:
Is instanceID the same as instance_index? Also why do we need this now?

Reply:
Also, this is only used once, is it worth having a method for it?

Reply:
So for v2 envelopes, the equivalent of v1envelope.InstanceIndex is either the `gauge.GetMetrics()["instance_index"]` (if it exists) or `v2envelope.InstanceId`. See e.g. https://github.com/cloudfoundry/go-loggregator/blob/master/conversion/tov1.go#L238